### PR TITLE
Dynamic time ranges added to the spy widget. It defaults to using epoch sec.

### PR DESCRIPTION
**Pull request by havlan**
Uses epoch_second now! It compares the two (current epoch millis) timestamps provided, and after that it computes time between in seconds and uses this in the query.

Further improvements would include extracting data from ```indexPattern``` to handle the most common time fields (eg '@timestamp').

This closes https://github.com/sirensolutions/sentinl/issues/454

### CODE REVIEW

**Comment by havlan** [comment]
#454

**Comment by sergibondarenko** [comment]
@havlan I'm reviewing this PR. Are you still working on it? If yes, I stop and continue when you finish it.

**Comment by havlan** [comment]
@sergibondarenko I'm satisfied. Sorry for not letting you know.

**Comment by havlan** [comment]
@sergibondarenko If anything it would be useful to let this work for other time "units", but I guess kibana defaults to epoch_millis.

**Comment by sergibondarenko** [comment]
> I'm satisfied. Sorry for not letting you know.

@havlan that's ok. Next time please add **WIP** (work in progress) prefix to PR's title.

File: public/dashboard_spy_button/alarm_button.js [hunk 1]
```diff
@@ -85,9 +103,30 @@ const dashboardSpyButton = function ($scope, config) {
                   let start = new Date(must[key].range['@timestamp'].gte);
                   let end = new Date(must[key].range['@timestamp'].lte);
                   let diff = new Date(end - start).getTime() / 1000; // UTC timestamp in seconds
-                  newTimestamp.gte = 'now-' + diff + 's/s';
-                  newTimestamp.lte = 'now/s';
-                  newTimestamp.format = must[key].range['@timestamp'].format.replace('millis','second');
+                  let timeArray = getUTCArray(diff,timeFractions);
+                  let largestUnit = 0;
+                  for(let i = 0; i < timeArray.length; ++i) {
+                    //[s,min,hour,day,week] == input array [60,60,24,7]
```
Comment (sergibondarenko) [inline public/dashboard_spy_button/alarm_button.js]: What does this comment mean? Delete if it is not needed?

File: public/dashboard_spy_button/alarm_button.js [hunk 2]
```diff
@@ -85,9 +103,30 @@ const dashboardSpyButton = function ($scope, config) {
...
+                        for(let g = k; g > largestUnit; --g) {
+                          timeSize *= timeFractions[g - 1];
+                        }
+                      }
+                      relativeTime += timeArray[k] * timeSize;
+                    }
+                  }
+                  newTimestamp.gte = 'now-' + relativeTime + timeUnits[largestUnit] + '/' + timeUnits[largestUnit];
+                  newTimestamp.lte = 'now/' + timeUnits[largestUnit];
+                  //newTimestamp.format = must[key].range['@timestamp'].format.replace('millis','second');
```
Comment (sergibondarenko) [inline public/dashboard_spy_button/alarm_button.js]: Delete this comment.

**Pull request merged**